### ISSUE TITLE
'gradlew tasks' fails since 'org.shipkit.compare-publications' is applied to shipkit build

### ISSUE THREAD

**Issue by epeee** [mockito/shipkit#498]
let's make it possible to execute `tasks` again.

In order to reproduce this one just execute `./gradlew tasks` on master.

```
~/shipkit$ ./gradlew tasks
  Building version '0.9.110' (value loaded from 'version.properties' file).
  [INCUBATING] upgrade-dependency plugin is incubating and may change in any version.
  [INCUBATING] downstream-testing plugin is incubating and may change in any version.
:tasks

------------------------------------------------------------
All tasks runnable from root project
------------------------------------------------------------

:tasks FAILED

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':tasks'.
> Could not determine the dependencies of task ':shipkit:comparePublications'.

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output.

BUILD FAILED

```

**Comment by wwilk** [comment]
There is a timing issue. Applying **comparePublicationsPlugin** adds a publication, that in turn creates **generatePomFileForJavaLibraryPublication** task. Seems that this task is created later than **comparePublications** task is configured, that's why there is this exception. I'm not sure yet why that is. I've investigated a little and two possible reasons come so far:
- **generate~** task is created with Gradle rules, I'm not sure when exactly they are executed, [here](https://github.com/gradle/gradle/blob/b76067a5183e358bd7bc56501eda8a74e88e1bb5/subprojects/maven/src/main/java/org/gradle/api/publish/maven/plugins/MavenPublishPlugin.java#L156) is the code
- seems that GradleDSLHelper delays the configuration of publications
@szczepiq Do you know what can be the reason? Can we somehow register a hook that would apply **comparePublications** after pom generation tasks are created?

**Comment by epeee** [comment]
This one is fixed now (#538).
